CMakeLists.txt

```diff
@@ -92,28 +113,98 @@ add_custom_target(toplevel_header
+                LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        )
 
-target_link_libraries(ecdaa
-        PRIVATE ${AMCL_LIBRARIES}
-        ${ECDAA_SEED_LINK_LIBRARY})
+        add_dependencies(ecdaa toplevel_header)
+endif (BUILD_SHARED_LIBS)
+
+################################################################################
+# Static Libary
+################################################################################
+if (BUILD_STATIC_LIBS)
+        add_library(ecdaa_static STATIC ${ECDAA_SRCS})
+
+        set_target_properties(ecdaa_static PROPERTIES
+                OUTPUT_NAME "ecdaa${STATIC_SUFFIX}"
+                VERSION "${ECDAA_VERSION}"
+                SOVERSION "${ECDAA_SOVERSION}"
+        )
+
+        target_include_directories(ecdaa_static PUBLIC
+                $<BUILD_INTERFACE:${GENERATED_TOPLEVEL_INCLUDE_DIR}>
+                $<BUILD_INTERFACE:${AMCL_INCLUDE_DIRS}>
+                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+        )
+
+        target_link_libraries(ecdaa_static
+                PRIVATE ${AMCL_LIBRARIES}
+                ${ECDAA_SEED_LINK_LIBRARY})
+
+        install(TARGETS ecdaa_static
+                EXPORT ${CMAKE_PROJECT_NAME}Targets
+                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+                LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        )
 
+        add_dependencies(ecdaa_static toplevel_header)
+endif (BUILD_STATIC_LIBS)
+
+################################################################################
+# Headers
+################################################################################
+install(DIRECTORY ${GENERATED_TOPLEVEL_INCLUDE_DIR}/ecdaa DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+install(FILES ${GENERATED_TOPLEVEL_INCLUDE_DIR}/ecdaa.h DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+
+################################################################################
+# pkgconfig
+################################################################################
+configure_file(ecdaa.pc.in ecdaa.pc @ONLY)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/ecdaa.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+################################################################################
+# CMake export
+################################################################################
+install(EXPORT ${CMAKE_PROJECT_NAME}Targets
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${CMAKE_PROJECT_NAME}
+        FILE        ${CMAKE_PROJECT_NAME}Config.cmake
+)
+
+################################################################################
+# Examples
+################################################################################
 if(ECDAA_BUILD_EXAMPLE_PROGRAMS)
         set(ECDAA_PROGRAMS_DIR examples)
         add_subdirectory(${ECDAA_PROGRAMS_DIR})
 endif(ECDAA_BUILD_EXAMPLE_PROGRAMS)
```

Comment:
The examples and test flag checks aren't consistent.  The `BUILDING_TESTING` flag is checked inside `test/CMakeLists.txt`.  `BUILD_EXAMPLES` is checked here before the `add_subdirectory(examples)` call.

I don't have a reason to prefer one or the other, but we should probably standardize.

Reply:
I've moved both to be in the top-level CMakeLists.txt. Good catch.